src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator/AbstractResxGenerator.cs

```diff
@@ -230,21 +229,21 @@ public bool Execute(GeneratorExecutionContext context)
                     _ => "cs",
                 };
 
-                OutputTextHintName = ResourceName + $".Designer.{extension}";
+                OutputTextHintName = ResourceInformation.ResourceName + $".Designer.{extension}";
 
-                if (string.IsNullOrEmpty(ResourceName))
+                if (string.IsNullOrEmpty(ResourceInformation.ResourceName))
                 {
                     LogError(language, "ResourceName not specified");
                     return false;
                 }
 
-                var resourceAccessName = RoslynString.IsNullOrEmpty(ResourceClassName) ? ResourceName : ResourceClassName;
+                var resourceAccessName = RoslynString.IsNullOrEmpty(ResourceInformation.ResourceClassName) ? ResourceInformation.ResourceName : ResourceInformation.ResourceClassName;
                 SplitName(resourceAccessName, out var namespaceName, out var className);
 
                 var classIndent = namespaceName == null ? "" : "    ";
                 var memberIndent = classIndent + "    ";
 
-                var text = ResourceFile.GetText(context.CancellationToken);
+                var text = ResourceInformation.ResourceFile.GetText(cancellationToken);
                 if (text is null)
```

Comment:
GitHub isn't letting me comment on the exact line because it's a bit far from the changes, but few lines below there is a `text.ToString()`

Can resx files be tooo large in a way that this call can hurt performance (ie, do LOH allocation)?

Is it worth implementing a custom `TextReader` that operates on the `SourceText` by reading small parts of it, assuming `XDocument` will properly try to read small parts at a time, avoiding the possible LOH allocations?

*I could be way too paranoid about LOH string allocations, but (as you may already know), I have seen them a lot in traces*

It all depends about how large resx files can get, and how many of those large files can be present.

Reply:
In this case, I don't think it'll make a difference. This generator will only re-run when the input changes. If it's easy to make the text reader it probably wouldn't hurt though.

Reply:
➡️ Now using `SourceTextReader`